### ISSUE TITLE
Allow mod files within tests directory

### ISSUE THREAD

**Issue by TomFryersMidsummer** [rust-lang/rust-clippy#11775]
### Summary

Within the `tests/` directory, `common/mod.rs` has an advantage over `common.rs` for shared code: it doesn't try to run `common.rs` as a test file, which would add noise to test runs. So, perhaps `mod_module_files` should exclude this case.

This Rust Book includes this recommendation: https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests.

### Lint Name

mod_module_files

### Reproducer

I tried this code:

```
├── Cargo.lock
├── Cargo.toml
├── src
│   └── lib.rs
└── tests
    ├── common
    │   └── mod.rs
    └── test.rs
```
with `mod common;` in `test.rs`. Other files are as generated by `cargo new`, or empty.
```sh
cargo clippy --tests -- -W clippy::mod-module-files
```
I saw this happen:

```
warning: `mod.rs` files are not allowed, found `tests/common/mod.rs`
 --> tests/common/mod.rs:0:1
  |
  |
  = help: move `tests/common/mod.rs` to `tests/common.rs`
  = note: requested on the command line with `-W clippy::mod-module-files`
```

I expected to see this happen:

[no error]

### Version

```text
rustc 1.73.0 (cc66ad468 2023-10-03)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: x86_64-unknown-linux-gnu
release: 1.73.0
LLVM version: 17.0.2
```

**Comment by miguelpduarte** [comment]
I have also faced this issue. Versions:
```
clippy 0.1.88 (6b00bc3880 2025-06-23)
rustc 1.88.0 (6b00bc388 2025-06-23)
```

This seems like the offending section: https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/module_style.rs#L153

For example for the workspace I'm working in, paths can be `crates/<some-crate>/tests/common/mod.rs`

Since the path starts with `crates/<some-crate>` instead of just `tests`, `!path.starts_with("tests")` is not fulfilled.

I guess it might make sense to either reopen this, or create a separate issue for tracking this specifically for workspace crates.
